Make QRcode decoding non-blocking by doing it in a thread

Cheaterman opened the pull request:
I noticed my entire application was frozen the first time I used zbarcam, this seems to fix it. :+1:

Cheaterman committed:
Make QRcode decoding non-blocking by doing it in a thread

AndreMiras reviewed (changes requested):
Thanks @Cheaterman for giving it a try.
It looks OK on the diff even though I haven't tried.
Note the unit tests are failing, could you try to update your PR fixing them.
The error is:
```
>       symbols = self.zbarcam._detect_qrcode_frame(texture, code_types)
E       TypeError: _detect_qrcode_frame() missing 1 required positional argument: 'code_types'
```
refs https://travis-ci.com/github/kivy-garden/zbarcam/jobs/371384314

Cheaterman commented:
Thanks for taking a look @AndreMiras - lemme fix that (I changed the signature of `_detect_qrcode_frame()`)

Cheaterman committed:
Fix _detect_qrcode_frame signature in tests

It takes pixels independently of texture because in normal usage the
decoding will be done in a separate thread (which therefore can't make
the required GL calls to access the pixels).

Cheaterman commented:
I think that will do :-)

Cheaterman committed:
Set _decoding_frame and _symbols_queue in __init__ for unit tests

Cheaterman commented:
Welp - this should be better. I had things set up in a way that wouldn't work if `_on_camera_ready` was never called - and it's not called in tests.

Cheaterman commented:
Haha, I'm silly, the tests need to run differently because of the now asynchronous nature of `_detect_qrcode_frame()` - I'll rewrite my code differently instead, leaving the function as it is, and using it inside of a separate threaded function.

Cheaterman committed:
Rewrite threading to make it test-friendly

Cheaterman commented:
It's "more or less" left as-is, since I need to pass the pixels separately (can't write to `texture.pixels`) ; I made it an optional kwarg and it uses `texture.pixels` by default.

Cheaterman committed:
Make isort happy as well.
